### ISSUE TITLE
API monitoring metrics

### ISSUE THREAD

**Issue by aniketmaurya** [Lightning-AI/LitServe#146]
## 🚀 Feature

Provide monitoring metrics: latency, decode-predict-encode time.

**Comment by aniketmaurya** [comment]
thanks for the feedback @carlthome! I would love to know more about your use case with the monitoring metrics.

**Comment by vrdn-23** [comment]
@aniketmaurya I think for any production use case we would want metrics that are similar to what we see in [TEI](https://github.com/huggingface/text-embeddings-inference) and [vllm](https://docs.vllm.ai/en/latest/serving/metrics.html#)

**Comment by AdolfoVillalobos** [comment]
Hello @aniketmaurya!

I took my chances on having a first stab at this issue in PR #262 

I left some things open-ended because I wanted to get some feedback from the team. After some clarification, I will add better test support and so on.

Also, I only registered simple metrics for the prediction elapsed time. The idea is to clarify the setup:

> @aniketmaurya I think for any production use case we would want metrics that are similar to what we see in [TEI](https://github.com/huggingface/text-embeddings-inference) and [vllm](https://docs.vllm.ai/en/latest/serving/metrics.html#)


Then, as the links showcase, we can extend the coverage of the metric collection to a broader set of metrics, including tokens per unit of time and so on.

As a side note, adding a benchmark routine to the CI might be a good idea. 

Cheers!
